src/transduce.jl

```diff
@@ -130,9 +130,9 @@ end
     arrays...,
 ) where {F,T}
     # basecase
-    n = length(idx)
+    n = @uniform length(idx)
     ill = @index(Local, Linear)
-    igl = @index(Group, Linear)
+    igl = @uniform @index(Group, Linear)
```

Comment:
```suggestion
    igl =  @index(Group, Linear)
```

Reply:
I thought I might need to put it in `@uniform` since I'm using it after `@synchronize`?

https://github.com/JuliaFolds/FoldsKernelAbstractions.jl/blob/18c7b8f96bc28569a26937ea5de27ebe23811135/src/transduce.jl#L173-L175

I can re-fetch the index, there, though

Reply:
`@index` expression are always valid (and automatically inserted into the loop body).

Reply:
Without `@uniform`, I'm now getting `UndefVarError: igl not defined`: https://github.com/JuliaFolds/FoldsKernelAbstractions.jl/pull/6/checks?check_run_id=1958277481

Actually, I just remember putting it in `@uniform` since it was the dependency of other `@uniform` variables:

https://github.com/JuliaFolds/FoldsKernelAbstractions.jl/blob/064aba9d053e9c1a26cdc5c59e3434c462d6cdb8/src/transduce.jl#L151-L152